Model/Indexer/Category.php

```diff
@@ -41,6 +43,9 @@ public function executeList(array $ids)
 
     public function executeRow($id)
     {
+        if (count(self::$affectedProductIds)) {
+            $this->categoryBatchQueueProcessor->setAffectedProductIds(self::$affectedProductIds);
```

Comment:
Would it make sense to just relocate this to the `\Algolia\AlgoliaSearch\Service\Category\BatchQueueProcessor` with a static setter/getter? Is there any benefit to shuffling around objects in this way? 

Also any reason to limit this check to only `executeRow`?

Reply:
I'm not fully sure if it's meaningful on a code POV indeed. For me this works because the products ids are added to the category indexer which is injected in the `CategoryObserver` so that we can make sure those are not "lost in the process" . I guess if we want to shunt the category indexer, we can inject directly the `BatchQueueProcessor` into the observer since this is the only place where this logic exists, I can try that.